File: old/java/230/Solution2.java
class Solution {
	private int cnt;
	private int ans;

	public int kthSmallest(TreeNode root, int k) {
		recur(root, k);
		return ans;
	}

	private void recur(TreeNode cur, int k) {
		if (cur == null) return;
		recur(cur.left, k);
		if (++cnt == k) {
			ans = cur.val;
			return;
		}
		recur(cur.right, k);
	}
}

File: archive/2021/cpp/226/solution2.cpp
#include <queue>


struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right)
        : val(x), left(left), right(right)
    {}
};


class Solution
{
public:
    TreeNode *invertTree(TreeNode *root)
    {
        std::queue<TreeNode *> q;
        if (root) q.push(root);
        while (!q.empty()) {
            TreeNode *cur = q.front();
            q.pop();
            if (cur->left) q.push(cur->left);
            if (cur->right) q.push(cur->right);
            std::swap(cur->left, cur->right);
        }
        return root;
    }
};
